frontend/public/components/namespace-dropdown.tsx

```diff
@@ -0,0 +1,309 @@
+import * as React from 'react';
+import {
+  Button,
+  Divider,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStatePrimary,
+  Menu,
+  MenuContent,
+  MenuGroup,
+  MenuInput,
+  MenuItem,
+  MenuList,
+  Switch,
+  TextInput,
+  Title,
+} from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { useUserSettingsCompatibility } from '@console/shared';
+import { isSystemNamespace } from './factory/table-filters';
+import NamespaceMenuToggle from './namespace-menu-toggle';
+
+const NamespaceBarDropdown_ = (props: NamespaceBarDropdownProps) => {
+  const { t } = useTranslation();
+  const {
+    actions,
+    disabled,
+    isProjects,
+    onSelect,
+    options,
+    selected,
+    shortCut,
+    storageKey,
+    title,
+    userSettingsPrefix,
+  } = props;
+
+  const MAX_HEIGHT = '60vh';
+
+  const filterRef = React.useRef(null);
+  const menuRef = React.useRef(null);
+
+  const [isMainSelectOpen, setMainSelectOpen] = React.useState(false);
+  const [filterText, setFilterText] = React.useState('');
+  const [activeOption, setActiveOption] = React.useState(selected);
+  const [footerHeight, setFooterHeight] = React.useState(0);
+
+  React.useEffect(() => {
+    setActiveOption(selected);
+  }, [selected, options]);
```

Comment:
I'm hesitant to use `options` in the dependency list. This could hot loop if the caller recalculates `options` on every render.

Reply:
See comment above.  I'm relooking at active option logic.